src/components/user.js

```diff
@@ -19,11 +21,52 @@ export const FIELD_NAMES = {
     website: 'website'
 };
 
+const SHOW = {
+    show: 'show',
+    hide: 'hide'
+};
+
+export const VIEWS = {
+    [MODES.editing]: {
+        [SHOW.show]:      {
+            buttonAddress: 'Show Address',
```

Comment:
to many code duplications here.
It's better to leave `this.state.showAddress` and `this.state.showCompany` boolean
and assign in `VIEWS` only e.g. `showPrefix` and `hidePrefix`